operator-pipeline-images/operatorcert/entrypoints/github_wait_labels.py

```diff
@@ -0,0 +1,205 @@
+
+    @staticmethod
+    def get_wait_conditions(args: ap.Namespace) -> list["WaitCondition"]:
+        """
+        Parse all required wait conditions from program args
+        """
+        conditions = []
+
+        for regexp in args.any:
+            conditions.append(WaitCondition(WaitType.WaitAny, regexp))
+
+        for regexp in args.none:
+            conditions.append(WaitCondition(WaitType.WaitNone, regexp))
+
+        return conditions
+
+    def __eq__(self, __value: object) -> bool:  # pragma: nocover
+        if not isinstance(__value, WaitCondition):
+            return False
+
+        return self.wait_type == __value.wait_type and self.regexp == __value.regexp
+
+
+def setup_argparser() -> ap.ArgumentParser:
+    """
+    Setup argument parser
+
+    Returns:
+        Any: Initialized argument parser
+    """
+    parser = ap.ArgumentParser(description="GitHub cli tool to wait on labels to PR.")
+
+    parser.add_argument(
+        "--any",
+        action="append",
+        default=[],
+        help="Wait until at least one label matches the regular expression.",
+    )
+
+    parser.add_argument(
+        "--none",
+        action="append",
+        default=[],
+        help="Wait until no labels match the regular expression.",
+    )
+
+    parser.add_argument(
+        "--github-host-url",
+        default="https://api.github.com",
+        help="The GitHub host, default: https://api.github.com",
+    )
+
+    parser.add_argument(
+        "--pull-request-url",
+        required=True,
+        help="The GitHub pull request URL where we want to wait for labels.",
+    )
+
+    parser.add_argument("--timeout", default=60, help="Timeout for waiting in seconds.")
+
+    parser.add_argument(
+        "--poll-interval", default=1, help="Interval between requests in seconds"
+    )
+
+    return parser
+
+
+def get_pr_labels(
+    repository: Repository,
+    pull_request_id: int,
+) -> list[str]:
+    """
+    Get labels on a PR in a repository by PR id
+    """
+    try:
+        pull_request: PullRequest = repository.get_pull(pull_request_id)
+    except GithubException as exc:
+        LOGGER.error("Unable to get pull request: %s", str(exc))
+        exit(1)
+
+    label_names = [label.name for label in pull_request.labels]
+
+    return label_names
+
+
+def wait_on_pr_labels(
+    repository: Repository,
+    pull_request_id: int,
+    wait_conditions: list[WaitCondition],
+    timeout_s: int,
+    poll_interval_s: int,
+) -> bool:
+    """
+    Wait until ALL wait conditions on PR labels hold
+
+    Args:
+        repository: PyGithub repository object
+        pull_request_id: Github pull request id
+        timeout_s: Timeout for waiting on label changes in seconds
+        wait_conditions: List of conditions that must hold for a wait to end
+        poll_interval_s: Interval between finished requests to GitHub API
+    """
+    assert poll_interval_s < timeout_s
```

Comment:
Might be good to add a human readable message as a second argument to this assert, smth like "timeout needs to be bigger than the poll interval"